### PR TITLE
Improuve Start script and fix stalwart 0.9.0

### PR DESCRIPTION
**Pull request by docjyJ**
Add new feature for personalisation.

### CODE REVIEW

**Comment by docjyJ** [comment]
What do you think about this PR @szaimen

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,284 @@
+#!/bin/bash
+
+if [ ! -f '/opt/stalwart-mail/etc/aio-config.env' ]; then
+  mkdir -p '/opt/stalwart-mail/etc'
+  echo '## REMOVE VARIABLES BELOW IF YOU WANT TO USE DEFAULT VALUES ##' > '/opt/stalwart-mail/etc/aio-config.env'
+fi
+
+source '/opt/stalwart-mail/etc/aio-config.env'
```
Comment (szaimen) [inline entrypoint.sh:8]: I guess here should be a check that checks if the file is there?

**Review by szaimen — COMMENTED** [review]
> What do you think about this PR @szaimen

I had a fast look at this and looks good in general (but did not test)

So it seems like you want to move the documentation for this container into the readme here? I guess you would need to adjust the link here then: https://github.com/nextcloud/all-in-one/blob/a41dc9f2c0bfdcee02b7d2e42fb1ee0ee778ad0a/community-containers/stalwart/stalwart.json#L6

Also,  I am not sure if providing bash scripts for upgrading is recommendable as they need to be executed as root most of the times. Maybe documentation like before makes more sense?

File: entrypoint.sh (hunk 1 above)
Comment (docjyJ) [reply entrypoint.sh]: 85e2d06

**Comment by docjyJ** [comment]
> I had a fast look at this and looks good in general (but did not test)

Don't worry!

> So it seems like you want to move the documentation for this container into the readme here? I guess you would need to adjust the link here then: https://github.com/nextcloud/all-in-one/blob/a41dc9f2c0bfdcee02b7d2e42fb1ee0ee778ad0a/community-containers/stalwart/stalwart.json#L6

Yes, nextcloud/all-in-one#5139

I want it to be as simple as possible for beginners.

I would like to expand the content based on feedback and my experience. Put links to resources to learn how SMTP/IMAP protocols work and all the magic words DMRAC, SFP, DKIM, DNS, MX...

I am learning a lot, that's mainly why I am building my own cloud with the RK3588 chip. (This could be an idea for a talk in a future community event: "How to be independent with your own low-power cloud?").

> Also, I am not sure if providing bash scripts for upgrading is recommendable as they need to be executed as root most of the times. Maybe documentation like before makes more sense?

Okay, I tried to separate the migration into simple steps. 2ff9c02

**Comment by szaimen** [comment]
> I want it to be as simple as possible for beginners.
> 
> I would like to expand the content based on feedback and my experience. Put links to resources to learn how SMTP/IMAP protocols work and all the magic words DMRAC, SFP, DKIM, DNS, MX...
> 
> I am learning a lot, that's mainly why I am building my own cloud with the RK3588 chip. (This could be an idea for a talk in a future community event: "How to be independent with your own low-power cloud?").

Sounds like a great idea! I would be very interested to watch such a talk :)

Btw, maybe this can be of some inspiration for you? https://github.com/szaimen/Nextcloud-NAS-Guide

> > Also, I am not sure if providing bash scripts for upgrading is recommendable as they need to be executed as root most of the times. Maybe documentation like before makes more sense?
> 
> Okay, I tried to separate the migration into simple steps. [2ff9c02](https://github.com/docjyJ/aio-stalwart/commit/2ff9c021050bcbc3bebed7230193cb37a2e4785e)

Looks good :)

Btw, maybe as idea, you could on xontainer stop do an export automatically which could ne imported if found after container update. We do such things in the database container for example: https://github.com/nextcloud/all-in-one/blob/a67fe51bbcd2091f0b49430c7564de0949c29cf2/Containers/postgresql/start.sh#L50-L196

**Comment by docjyJ** [comment]
> Sounds like a great idea! I would be very interested to watch such a talk :)

I'm going to try to present at a conference in France this year. If it goes well, I'll work on my English to present at a NextCloud event!

> Btw, maybe this can be of some inspiration for you? https://github.com/szaimen/Nextcloud-NAS-Guide

I didn't know, interesting, thanks for sharing the link.

> Btw, maybe as idea, you could on container stop do an export automatically which could be imported if found after container update.

Thanks, I'll look at that and if it's worth trying I'll open an issue.



I will merge this PR, test it on my instance and publish it this week or next.

Thank you so much,

**Comment by szaimen** [comment]
> > Sounds like a great idea! I would be very interested to watch such a talk :)
> 
> I'm going to try to present at a conference in France this year. If it goes well, I'll work on my English to present at a NextCloud event!

Sounds good :)

> > Btw, maybe this can be of some inspiration for you? https://github.com/szaimen/Nextcloud-NAS-Guide
> 
> I didn't know, interesting, thanks for sharing the link.

You are welcome :)

> > Btw, maybe as idea, you could on container stop do an export automatically which could be imported if found after container update.
> 
> Thanks, I'll look at that and if it's worth trying I'll open an issue.
> 
> I will merge this PR, test it on my instance and publish it this week or next.

Sounds good!

> Thank you so much,

No problem, you are welcome! :)

**Pull request merged**